state/machine_linklayerdevices.go

```diff
@@ -1076,9 +1072,85 @@ func (m *Machine) SetContainerLinkLayerDevices(containerMachine *Machine) error
 		// something useful.
 		containerSpaces, err = m.inferContainerSpaces(containerMachine.Id(), defaultSpaceName)
 		if err != nil {
-			return errors.Trace(err)
+			return nil, errors.Trace(err)
 		}
 	}
+	return containerSpaces, nil
+}
+
+// FindMissingBridgesForContainer looks at the spaces that the container
```

Comment:
Does this belong in the state package? can we move it to apiserver/common/<networkcommon> or so? It looks like we would need to expose Machine.determineContainerSpaces; is that unreasonable?